libraries/node-core-library/src/LockFile.ts

```diff
@@ -257,14 +257,23 @@ export class LockFile {
 
     let lockFile: LockFile;
 
-    try {
-      // open in write mode since if this file exists, it cannot be from the current process
-      // TODO: This will malfunction if the same process tries to acquire two locks on the same file.
-      // We should ideally maintain a dictionary of normalized acquired filenames
-      lockFileHandle = FileWriter.open(pidLockFilePath);
-      lockFileHandle.write(startTime);
+    let currentBirthTimeMs: number;
 
-      const currentBirthTimeMs: number = FileSystem.getStatistics(pidLockFilePath).birthtime.getTime();
+    try {
+      try {
+        // open in write mode since if this file exists, it cannot be from the current process
+        // TODO: This will malfunction if the same process tries to acquire two locks on the same file.
+        // We should ideally maintain a dictionary of normalized acquired filenames
+        lockFileHandle = FileWriter.open(pidLockFilePath);
+        lockFileHandle.write(startTime);
+        currentBirthTimeMs = FileSystem.getStatistics(pidLockFilePath).birthtime.getTime();
```

Comment:
I would expect part of the problem to be that this is opening a second file handle to the same file in the same process.
We should consider adding a `getStatistics()` method to the `FileWriter` instance that returns `fsx.fstatSync(this._fileDescriptor)`, then call `lockFileHandle.getStatistics()` instead of `FileSystem.getStatistics(pidLockFilePath)`

Reply:
Thanks for the idea! I’m not sure it’s the root cause of the problem, but it definitely makes sense in terms of overall efficiency and simplicity of file system operations. I’ve added the `getStatistics` method to `FileWriter` class and reused `lockFileHandle` to get stats.